sqlglot/dialects.py

```diff
@@ -461,25 +628,27 @@ def _unix_to_time(self, expression):
 Hive.functions = {
     "APPROX_COUNT_DISTINCT": exp.ApproxDistinct.from_arg_list,
     "COLLECT_LIST": exp.ArrayAgg.from_arg_list,
-    "DATE_ADD": lambda args: exp.TsOrDsAdd(
+    "DATE_ADD": lambda args: exp.MixedTypeAdd(
         this=list_get(args, 0),
         expression=list_get(args, 1),
         unit=exp.Literal.string("DAY"),
+        output_format=exp.Literal.string("YYYY-MM-DD"),
     ),
     "DATEDIFF": lambda args: exp.DateDiff(
-        this=exp.DateStrToDate(this=list_get(args, 0)),
-        expression=exp.DateStrToDate(this=list_get(args, 1)),
+        this=exp.MixedTypeToDate(this=list_get(args, 0)),
+        expression=exp.MixedTypeToDate(this=list_get(args, 1)),
     ),
-    "DATE_SUB": lambda args: exp.TsOrDsAdd(
+    "DATE_SUB": lambda args: exp.MixedTypeAdd(
         this=list_get(args, 0),
         expression=exp.Mul(
             this=list_get(args, 1),
             expression=exp.Literal.number(-1),
         ),
         unit=exp.Literal.string("DAY"),
+        output_format=exp.Literal.string("YYYY-MM-DD"),
```

Comment:
i wonder if you need to be calling format_time in many places in this PR

format_time allows us to take in an input format string like hive "YYYY-MM" and convert it to presto "%Y-%M"